File: kp 2020 cpp/l3/Circle.cpp
/**
 * Creates circle
 * It extends virtual class Figure and inheritance methods surface and circumference
 *
 * @author Tobiasz Wojnar
 * excise for university course
 * @see Main.cpp for full requirements
 * @see Circle.hpp
 * @see Figure.hpp
 */
#include "Circle.hpp"
#include <cmath>

/**
 * Constructor of circle, since we create circle it needs only one parameter
 * @param radius is positive, double precision number
 */
Circle :: Circle (double radius){
    this->radius=radius;
}
/**
 * @return surface area of circle
 */
double Circle :: Surface () {
    return radius*radius*M_PI;
}
/**
 * @return circumference of circle
 */
double Circle :: Circumference () {
    return radius*2*M_PI;
}

File: l3/java/Circle.java
/**
 * Creates circles
 * It extends abstract class Figure and inheritance methods surface and circumference
 *
 * @author Tobiasz Wojnar
 * execise for university course
 * @see l3.Figury for full requriments
 * @see l3.Figure
 */
public class Circle extends Figure {
    double radius;

    /**
     * Constructor of circle, requires only one parameter
     * @param radius of circle, is positive, double precision number
     */
    public Circle (double radius){//TODO how to create exception if number is not positive
        this.radius = radius;
    }

    /**
     * @return surface area of circle
     */
    public double surface (){
        return Math.PI* radius * radius;
    }

    /**
     * @return circumference of circle
     */
    public double circumference (){
        return Math.PI* radius *2;
    }
}
